src/function/maybe_changed_after.rs

```diff
@@ -424,62 +464,74 @@ where
             return true;
         }
 
+        let verified_at = memo.verified_at.load();
+        let current_revision = zalsa.current_revision();
+
         // SAFETY: We do not access the query stack reentrantly.
         unsafe {
             zalsa_local.with_query_stack_unchecked(|stack| {
                 cycle_heads.iter().all(|cycle_head| {
-                    stack
-                        .iter()
-                        .rev()
-                        .find(|query| query.database_key_index == cycle_head.database_key_index)
-                        .map(|query| query.iteration_count())
-                        .or_else(|| {
-                            // If the cycle head isn't on our stack because:
-                            //
-                            // * another thread holds the lock on the cycle head (but it waits for the current query to complete)
-                            // * we're in `maybe_changed_after` because `maybe_changed_after` doesn't modify the cycle stack
-                            //
-                            // check if the latest memo has the same iteration count.
+                    // If the memo was verified in the current revision, check if it has the same iteration count as any query on the stack.
+                    // We skip this check for memo's from past revisions because we want to re-execute them if the cycle heads re-execute.
+                    if current_revision == verified_at {
```

Comment:
The main change here is that we skip the stack if `verified_at` isn't the current revision because we want all cycle participating queries to re-execute if they're from a past revision and their cycle head is re-executing